### PR TITLE
Server skeleton

### PR DESCRIPTION
**Pull request by mechina-19**

### CODE REVIEW

File: server/__pycache__/server.cpython-312.pyc
Comment (olivecylinder) [inline server/__pycache__/server.cpython-312.pyc:1]: should not be committed, add a gitignore

File: server/server.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+
+# @app.get("/lines/{line_id, station_id}")
+# def get_line_info(line_id: int, station_id:int):
+#     """
+#     Retrieve data about a given line: the stations it passes through and its departure times-table 
+#     """
+
+#     # todo: write the function
+
```
Comment (olivecylinder) [inline server/server.py:34]: remove or uncomment?

File: server/server.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    return {"Arriving": "12:54:32", "Location": {"Lat": "33.4234242", "Lon": "34.334211"}}
+
+@app.post("/passenger/wait-for/")
+def passenger_wait_for_bus(station_id: str, bus_id: str):
+    """
+    Log that a passenger is waiting for a specific bus at a given station.
+    """
+
+    # Store the entry (in-memory for now)
+    #waiting_passengers.append(station_id, bus_id))
```
Comment (olivecylinder) [inline server/server.py:61]: remove or uncomment?

File: server/server.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+
+@app.get("/update-station-list/{last_updated_date}")
+def update_station_list(last_updated_date: str):
+    """
+    Check whether or not the station list is up to date, and if not sending changes
+    """
+     # Compare last_updated_date to last date the station list was updated
+     # If equal: return {"status": "Up to date"}
+     # else: return {"status": "Not up to date", "changes": []}
```
Comment (olivecylinder) [inline server/server.py:92]: remove or uncomment? no return value

File: server/server.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    Check if the username and the password can be register, and if so register
+    """
+    return {"message", "200 OK"}
+
+@app.post("/passenger/signup")
+def passenger_signup(username: str, password: str):
+    """
+    Check if the username and the password can be register, and if so register
+    """
+    return {"message", "200 OK"}
```
Comment (olivecylinder) [inline server/server.py:120]: consider a unified login for both types? frontend (@number10mechina) says it's one screen

File: server/server.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+     # Compare last_updated_date to last date the station list was updated
+     # If equal: return {"status": "Up to date"}
+     # else: return {"status": "Not up to date", "changes": []}
+
+@app.post("/driver/login")
+def driver_login(username: str, password: str):
+    """
+    Check if the username and the password are correct
+    """
+    return {"message", "200 OK"}
```
Comment (olivecylinder) [inline server/server.py:99]: talk with frontend about a schema for the login API, it should return some user data as well

**Review by olivecylinder — CHANGES_REQUESTED** [review]

**Comment by olivecylinder** [comment]
[server skeleton](https://trello.com/c/DQbRGwjq/13-server-skeleton)

File: server/__pycache__/server.cpython-312.pyc
Comment (mechina-19) [reply server/__pycache__/server.cpython-312.pyc:1]: Added gitignore

File: server/server.py (hunk 3 above)
Comment (mechina-19) [reply server/server.py:92]: Added a place holder return in the meantime

File: server/server.py (hunk 4 above)
Comment (mechina-19) [reply server/server.py:120]: removed unnecessary functions

File: server/server.py (hunk 2 above)
Comment (mechina-19) [reply server/server.py:61]: leave it like this for now, when we will have the queue we would uncomment

File: server/server.py (hunk 5 above)
Comment (mechina-19) [reply server/server.py:99]: after thorough discussion it was decided it is unnecessary.

File: server/server.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
+from fastapi import FastAPI
+from pydantic import BaseModel
+
+app = FastAPI()
+class Station(BaseModel):
+    id: int
+    name: str
+    cords: tuple
```
Comment (olivecylinder) [inline server/server.py:8]: coordinates:)

File: server/server.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    line1 = {"line_id": 45634345, "bus_id": 1234, "Arrival Time": "12:41:23"}
+    line2= {"line_id": 4242234, "bus_id": 4432, "Arrival Time": "22:21:43"}
+    return {"lines":[line1, line2]}
+
+@app.get(" /update-arrival-time/{station_id, bus_id}")
+def update_arrival_time(station_id: int, bus_id: int):
+    """
+    Get real-time bus location and arrival time to a given station
+    """
+
```
Comment (olivecylinder) [inline server/server.py:42]: empty line between doc and code or not?

File: server/server.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    return {"message": "Passenger wait request logged successfully"}
+
+@app.post("/driver/drive/")
+def register_for_line(user_id: str, line_id: str, dep_time: str):
+    """
+    Register a driver for a specific line
+    """
+
+    # change the status of nahagos in this specific line
```
Comment (olivecylinder) [inline server/server.py:62]: if a todo, add `TODO:`

File: server/server.py [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+    return {"message": "Line registered successfully"}
+
+@app.delete("/driver/drive")
+def delete_drive(user_id: str, line_id: str, dep_time: str):
+    """
+    Delete a drive 
+    """
+    # change the status of nahagos in this specific line
```
Comment (olivecylinder) [inline server/server.py:71]: if a todo, add `TODO:`

File: server/server.py [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+    return {"message": "Drive was deleted successfully"}
+
+
+@app.get("/update-station-list/{last_updated_date}")
+def update_station_list(last_updated_date: str):
+    """
+    Check whether or not the station list is up to date, and if not sending changes
+    """
+     # Compare last_updated_date to last date the station list was updated
```
Comment (olivecylinder) [inline server/server.py:81]: if a todo, add `TODO:`

File: server/server.py (hunk 5 above)
Comment (olivecylinder) [reply server/server.py:93]: > it was decided it is unnecessary.

it was deemed*

File: server/server.py [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+def get_lines_starting(station_id: int):
+    """
+    Retrieve a list of bus lines that depart from a given station.
+    """
+    line1 = {"line_id": 45634345}
+    line2= {"line_id": 12341523}
+    return {"lines":[line1, line2]}
+
+
+
```
Comment (olivecylinder) [inline server/server.py:26]: consistent spacing between functions
(reformat before committing?)

**Review by olivecylinder — CHANGES_REQUESTED** [review]

**Review by olivecylinder — APPROVED** [review]

**Pull request merged**